dashboard/visualization/z_score_plot.py

```diff
@@ -0,0 +1,171 @@
+    plot_control_pos_df = prepare_data_plot(
+        control_pos_df, WELL, Z_SCORE, range_max=range_max
+    )
+    plot_control_neg_df = prepare_data_plot(
+        control_neg_df, WELL, Z_SCORE, range_max=range_max
+    )
+
+    # to be updated
+    dfs = [plot_control_pos_df, plot_control_neg_df, plot_compounds_df]
+    colors = ["rgb(0,128,0)", "rgb(255,0,0)", "rgb(31, 119, 180)"]
+    colors_shade = ["rgba(0,128,0,0.2)", "rgba(255,0,0,0.2)", "rgba(31, 119, 180,0.4)"]
+    names = ["CONTROL POS", "CONTROL NEG", "COMPOUNDS"]
+
+    for i, df in enumerate(dfs):
+        fig.add_trace(
+            go.Scatter(
+                x=df[id_pos],
+                y=df["mean"],
+                line=dict(color=colors[i]),
+                mode="lines",
+                name=names[i],
+                legendgroup=names[i],
+                customdata=np.stack((df[WELL], df["std"]), axis=-1),
+                hovertemplate="well: %{customdata[0]}<br>avg z-score: %{y:.2f} &plusmn;%{customdata[1]:.2f}<extra></extra>",
+            )
+        )
+
+        fig.add_trace(
+            go.Scatter(
+                x=pd.concat(
+                    [df[id_pos], df[id_pos][::-1]], ignore_index=True
+                ),  # well, then well reversed
+                y=pd.concat(
+                    [df["ub"], df["lb"][::-1]], ignore_index=True
+                ),  # upper, then lower reversed
+                fill="toself",
+                fillcolor=colors_shade[i],
+                line=dict(color="rgba(255,255,255,0)"),
+                hoverinfo="skip",
+                showlegend=False,
+                name="Shaded Area",
+                legendgroup=names[i],
+            )
+        )
+
+    if compounds_outside_df is not None:
+        compounds_outside_df = compounds_outside_df.merge(
+            plot_compounds_df[[WELL, id_pos]], on=WELL
+        )
+        fig.add_trace(
+            go.Scatter(
+                x=compounds_outside_df[id_pos],
+                y=compounds_outside_df[Z_SCORE],
+                mode="markers",
+                marker=dict(color="rgb(31, 119, 180)", size=8),
+                name="COMPOUNDS OUTSIDE",
+                customdata=np.stack(
+                    (compounds_outside_df[PLATE], compounds_outside_df[WELL]), axis=-1
+                ),
+                hovertemplate="plate: %{customdata[0]}<br>well: %{customdata[1]}<br>z-score: %{y:.2f}<extra>CMPD ID</extra>",
+            )
+        )
+    else:
+        fig.add_trace(
+            go.Scatter(
+                x=list(),
+                y=list(),
+                mode="markers",
+                marker=dict(color="rgb(31, 119, 180)", size=8),
+                name="COMPOUNDS OUTSIDE",
+            )
+        )
+
+    fig.add_hline(
+        y=z_score_limits[0],
+        line_width=3,
+        line_dash="dash",
+        line_color="gray",
+        annotation_text=f"MIN: {z_score_limits[0]:.2f}",
+        annotation_font_color="gray",
```

Comment:
the current styling makes it difficult to see the annotation when a lot of points are on the plot:

image.png